listen/property-changes.js

```diff
@@ -57,41 +55,95 @@ module.exports = PropertyChanges;
 function PropertyChanges() {
     throw new Error("This is an abstract interface. Mix it. Don't construct it");
 }
+var PropertyChangesP = PropertyChanges.prototype;
+
+function ObjectPropertyChangeDescriptor(propertyName) {
+	this.propertyName = propertyName;
+	return this;
+}
+Object.defineProperties(ObjectPropertyChangeDescriptor.prototype,{
+	propertyName: {
+		value:null,
+		writable: true
+	},
+	_willChangeListeners: {
+		value:null,
+		writable: true
+	},
+	willChangeListeners: {
+		get: function() {
+			return this._willChangeListeners || (this._willChangeListeners = new WillChangeListenersRecord(this.propertyName));
+		}
+	},
+	_changeListeners: {
+		value:null,
+		writable: true
+	},
+    changeListeners: {
+		get: function() {
+			return this._changeListeners || (this._changeListeners = new ChangeListenersRecord(this.propertyName));
+		}
+	}
+});
```

Comment:
Getters and setters might be an unnecessary overhead for this private interface. Declare all possible private fields as null in the constructor to benefit from hidden classes in V8. Create records on demand at each call site or in methods.